The REST API documentation /api/events POST endpoint returns status 500

Using the following metadata for creating a new scheduled event, the event get's created when using `curl`, yet when using the REST API documentation page it returns an https status 500.

```bash
#!/bin/sh

metadata='[
  {
    "flavor":"dublincore/episode",
    "fields":[
      {
        "id":"title",
        "value":"Test Livestream"
      },
      {
        "id":"subjects",
        "value":
          [
            "test",
          ]
      },
      {
        "id":"creator",
        "value":["Kristof"]
      },
      {
        "id":"description",
        "value":"test"},
    ]
  }
]'

scheduling='
  {
    "agent_id":"GCMobile-ocrec40",
    "start":"2021-12-14T09:28:00Z",
    "end":"2021-12-14T10:28:00Z",
    "inputs":["defaults"]
  }'

acl='[
  {
    "allow":true,
    "role":"ROLE_ADMIN",
    "action":"read"},
  {
    "allow":true,
    "role":"ROLE_ADMIN",
    "action":"write"
  }
]'

curl -u admin:opencast \
  -v \
  -X POST \
	http://localhost:8080/api/events \
	-F acl="$acl" \
	-F metadata="$metadata" \
	-F processing='{"workflow": "fast"}' \
	-F scheduling="$scheduling"
```

The error that gets logged : 

```
2021-12-10 13:18:00,344 | ERROR | (EventsEndpoint:627) - Unable to create event
org.opencastproject.index.service.exception.IndexServiceException: Unable to parse new event.
	at org.opencastproject.index.service.impl.util.EventHttpServletRequest.createFromHttpServletRequest(EventHttpServletRequest.java:183) ~[!/:?]
	at org.opencastproject.external.endpoint.EventsEndpoint.createNewEvent(EventsEndpoint.java:604) [!/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179) [!/:3.4.3]
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) [!/:3.4.3]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201) [!/:3.4.3]
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104) [!/:3.4.3]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) [!/:3.4.3]
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96) [!/:3.4.3]
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308) [!/:3.4.3]
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) [!/:3.4.3]
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:265) [!/:3.4.3]
	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234) [!/:3.4.3]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208) [!/:3.4.3]
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160) [!/:3.4.3]
	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:225) [!/:3.4.3]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:298) [!/:3.4.3]
	at org.opencastproject.kernel.rest.RestPublisher$RestServlet.handleRequest(RestPublisher.java:559) [!/:?]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPost(AbstractHTTPServlet.java:217) [!/:3.4.3]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707) [!/:3.1.0]
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:273) [!/:3.4.3]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:763) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1631) [!/:9.4.28.v20200408]
	at org.opencastproject.kernel.rest.JsonpFilter.doFilter(JsonpFilter.java:113) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) [!/:9.4.28.v20200408]
	at org.opencastproject.kernel.filter.proxy.TransparentProxyFilter.doFilter(TransparentProxyFilter.java:63) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) [!/:9.4.28.v20200408]
	at org.opencastproject.security.urlsigning.filter.UrlSigningFilter.doFilter(UrlSigningFilter.java:86) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) [!/:9.4.28.v20200408]
	at org.opencastproject.kernel.security.RemoteUserAndOrganizationFilter.doFilter(RemoteUserAndOrganizationFilter.java:247) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) [!/:9.4.28.v20200408]
	at org.opencastproject.kernel.rest.CurrentJobFilter.doFilter(CurrentJobFilter.java:87) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) [!/:9.4.28.v20200408]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:118) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:84) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.opencastproject.kernel.security.AsyncTimeoutRedirectFilter.doFilter(AsyncTimeoutRedirectFilter.java:60) [!/:?]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:113) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:103) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:113) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:146) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:54) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:45) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:98) [!/:3.1.4.RELEASE]
	at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilter(BasicAuthenticationFilter.java:150) [!/:3.1.7.RELEASE]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
	at org.springframework.security.web.authentication.www.DigestAuthenticationFilter.doFilter(DigestAuthenticationFilter.java:115) [!/:3.1.7.RELEASE]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
	at org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:82) [!/:3.1.4.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.ui.DefaultLoginPageGeneratingFilter.doFilter(DefaultLoginPageGeneratingFilter.java:91) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:183) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:105) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192) [!/:3.1.7.RELEASE]
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160) [!/:3.1.7.RELEASE]
	at org.opencastproject.kernel.security.SecurityFilter.doFilter(SecurityFilter.java:124) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) [!/:9.4.28.v20200408]
	at org.opencastproject.kernel.security.OrganizationFilter.doFilter(OrganizationFilter.java:135) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) [!/:9.4.28.v20200408]
	at org.opencastproject.kernel.filter.https.HttpsFilter.doFilter(HttpsFilter.java:73) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) [!/:9.4.28.v20200408]
	at org.opencastproject.kernel.rest.CleanSessionsFilter.doFilter(CleanSessionsFilter.java:89) [!/:?]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:226) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1618) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:549) [!/:9.4.28.v20200408]
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:71) [!/:?]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1363) [!/:9.4.28.v20200408]
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:298) [!/:?]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:489) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1278) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [!/:9.4.28.v20200408]
	at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:82) [!/:?]
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.Server.handle(Server.java:500) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806) [!/:9.4.28.v20200408]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938) [!/:9.4.28.v20200408]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.nio.file.FileSystemException: ***/opencast-dist-allinone/data/opencast/files/mediapackage/533ea202-1801-49da-be5f-9c2f00b2442a/369e7245-748d-4def-8fe4-53b5ff133727/369e7245-748d-4def-8fe4-53b5ff133727959809063175720271.tmp -> ***/opencast-dist-allinone/data/opencast/files/mediapackage/533ea202-1801-49da-be5f-9c2f00b2442a/369e7245-748d-4def-8fe4-53b5ff133727: Is a directory
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:100) ~[?:?]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) ~[?:?]
	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:417) ~[?:?]
	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:267) ~[?:?]
	at java.nio.file.Files.move(Files.java:1422) ~[?:?]
	at org.opencastproject.workingfilerepository.impl.WorkingFileRepositoryImpl.put(WorkingFileRepositoryImpl.java:382) ~[?:?]
	at org.opencastproject.ingest.impl.IngestServiceImpl.addContentToRepo(IngestServiceImpl.java:1627) ~[?:?]
	at org.opencastproject.ingest.impl.IngestServiceImpl.addTrack(IngestServiceImpl.java:730) ~[?:?]
	at org.opencastproject.ingest.impl.IngestServiceImpl.addTrack(IngestServiceImpl.java:706) ~[?:?]
	at org.opencastproject.index.service.impl.util.EventHttpServletRequest.ingestFile(EventHttpServletRequest.java:208) ~[!/:?]
	at org.opencastproject.index.service.impl.util.EventHttpServletRequest.createFromHttpServletRequest(EventHttpServletRequest.java:173) ~[!/:?]
	... 114 more
```

_EDIT(Lars): code formatting_

--- a/modules/index-service/src/main/java/org/opencastproject/index/service/impl/util/EventHttpServletRequest.java
+++ b/modules/index-service/src/main/java/org/opencastproject/index/service/impl/util/EventHttpServletRequest.java
@@ -170,7 +170,11 @@ public class EventHttpServletRequest {
           if (item.isFormField()) {
             setFormField(eventCatalogUIAdapters, eventHttpServletRequest, item, fieldName, startDatePattern, startTimePattern);
           } else {
-            ingestFile(ingestService, eventHttpServletRequest, item);
+            if (!item.getName().isBlank()) {
+              ingestFile(ingestService, eventHttpServletRequest, item);
+            } else {
+              logger.debug("Skipping field {} due to missing filename", item.getFieldName());
+            }
           }
         }
       } else {

